### PR TITLE
Initial release process document

### PR DESCRIPTION
**Pull request by igaray**
# Release process documentation

## Description
- [x] Add `RELEASE.md` 
- [x] Update pull request template to include an action item to update the changelog. This is now part of documentation process and PR acceptance criteria.
- [x] Update `CHANGELOG.md` to bring it back up to date.

## Checklist
- [ ] Linked to Github Issue
- [ ] Unit tests added
- [ ] Integration tests added.
- [x] This change requires new documentation.
  - [x] Documentation has been added/updated.

### CODE REVIEW

File: RELEASE.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+      The release string should have the format "vX.Y.Z", with a possible
+      trailing "-rcN" and follow [semantic versioning](https://semver.org/).
+- [ ] Checkout branch named `release-N` where N is the version string.
+- [ ] Update the version field in the package entry of `Cargo.toml` files.
+  - The versions must be the same.
+  - There are 4 relevant `Cargo.toml` files in the repo:
+    - `Cargo.toml`: update the version string.
+    - `cairo-vm-cli/Cargo.toml`: update the version string and also the `cairo-vm` dependency version to match the above.
+    - `felt/Cargo.toml`: update the version string.
+    - `deps/parse-hyperlinks/Cargo.toml`: this vendored dependency needs its version bumped, but does not need to match the other crate versions.
```
Comment (Oppen) [inline RELEASE.md:14]: I think it's better to just split this, as this becomes busywork. I'll try to create a new repo for it this week (as "when I have time to send a patch to nom" seems not to happen) so we can update it separately and not need bumping it every time.

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -10,3 +10,4 @@ Description of the pull request changes and motivation.
 - [ ] Integration tests added.
 - [ ] This change requires new documentation.
   - [ ] Documentation has been added/updated.
+  - [ ] Changelog has been updated.
```
Comment (Oppen) [inline .github/pull_request_template.md:13]: This should suffice for now. A later refinement would be a pipelines that rejects the merge if it wasn't.

**Comment by igamigo** [comment]
Probably not very useful, but here are some takes that might be worth considering for the PR template:

- Verified that related documentation/references are still valid (eg, if the project structure changes and a readme is moved, are references from the documentation in `starknet-in-rust` still valid?). We might want to communicate these changes to pertinent teams.
- If dependencies are introduced/removed in a release, there might be some things that might have to be double checked (eg, is it no_std? is it cross-platform?). These are probably more related to the integration pipelines though.

File: RELEASE.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  - [Here](https://github.com/lambdaclass/cairo-rs/pull/948/files) is an example pull request with these changes.
+- Update `CHANGELOG.md`:
+  - Verify that the changelog is up to date.
+  - Add a title with the release version string just below the `Upcoming Changes` section.
+- [ ] Commit your changes, push your branch, and create a pull request.
+- [ ] Merge after CI and review passes.
+- [ ] Pull latest from `main` again.
+- [ ] Tag commit with version string and push tag.
```
Comment (igamigo) [inline RELEASE.md:22]: Something that could be useful here, and in the other steps, is to write the command that should be performed in order to make it more accessible (for example, add `git tag -a v1 -m "Release... ` to this step).

**Review by Oppen — APPROVED** [review]

**Review by pefontana — APPROVED** [review]

**Pull request merged**